pkg/util/resize/configspec.go

```diff
@@ -265,8 +265,8 @@ func compareExtraConfig(
 		return
 	}
 
-	extraConfig := util.ExtraConfigToMap(cs.ExtraConfig)
-	outCS.ExtraConfig = util.MergeExtraConfig(ci.ExtraConfig, extraConfig)
+	extraConfig := pkgutil.OptionValuesFromList(ci.ExtraConfig)
+	outCS.ExtraConfig = extraConfig.DiffListToList(cs.ExtraConfig)
```

Comment:
Nope, because of the ordering. Please note the comments on the functions. The `ToList` variants specifically maintain insertion order at the request of @bryanv to make ConfigSpec comparisons easier for resize.

Reply:
curious: Are we sure that by insertion ordering here, we also maintain the order that we get back from the configInfo when fetched from the VM?.

Reply:
There are some situations for writing simpler test assertions a deterministic ordering is helpful, but consider that ask secondary to whatever we want to have clearer non-test code.